src/aind_slims_api/core.py

```diff
@@ -212,35 +132,85 @@ def fetch_models(
         start: Optional[int] = None,
         end: Optional[int] = None,
         **kwargs,
-    ) -> tuple[list[SlimsBaseModelTypeVar], list[dict[str, Any]]]:
+    ) -> list[SlimsBaseModelTypeVar]:
         """Fetch records from SLIMS and return them as SlimsBaseModel objects
 
         Returns
         -------
         tuple:
             list:
                 Validated SlimsBaseModel objects
-            list:
-                Dictionaries representations of objects that failed validation
+
+        Notes
+        -----
+        - kwargs are mapped to field alias values
         """
+        resolved_kwargs = deepcopy(model._base_fetch_filters)
+        for name, value in kwargs.items():
+            resolved_kwargs[self.resolve_model_alias(model, name)] = value
+        logger.debug("Resolved kwargs: %s", resolved_kwargs)
+        resolved_sort: Optional[str | list[str]] = None
+        if sort is not None:
+            if isinstance(sort, str):
+                resolved_sort = self.resolve_model_alias(model, sort)
+            else:
+                resolved_sort = [
+                    self.resolve_model_alias(model, sort_key) for sort_key in sort
+                ]
+        logger.debug("Resolved sort: %s", resolved_sort)
         response = self.fetch(
-            model._slims_table.default,  # TODO: consider changing fetch method
+            model._slims_table,  # TODO: consider changing fetch method
```

Comment:
A warning about accessing a protected attribute. We should probably make it public or expose a public method to fetch this attribute.